pkg/k8sutil/k8sutil.go

```diff
@@ -189,3 +230,15 @@ func mergeOwnerReferences(old []metav1.OwnerReference, new []metav1.OwnerReferen
 	}
 	return old
 }
+
+func mergeMetadata(new *metav1.ObjectMeta, old metav1.ObjectMeta) {
+	new.SetLabels(mergeMaps(new.Labels, old.Labels))
+	new.SetAnnotations(mergeMaps(new.Annotations, old.Annotations))
+}
+
+func mergeMaps(new map[string]string, old map[string]string) map[string]string {
+	for k, v := range new {
+		old[k] = v
```

Comment:
This function probably needs a check for `if old == nil`?

Reply:
Yes, for sure. Thanks!